staging/src/k8s.io/client-go/rest/watch/decoder_test.go

```diff
@@ -54,7 +54,7 @@ func TestDecoder(t *testing.T) {
 		go func() {
 			data, err := runtime.Encode(scheme.Codecs.LegacyCodec(v1.SchemeGroupVersion), expect)
 			if err != nil {
-				t.Fatalf("Unexpected error %v", err)
+				t.Errorf("Unexpected error %v", err)
```

Comment:
The `Fatalf` call exits early but the `Errorf`does not, so if there is an error it will be printed and then there will be a nil pointer exception on line 61, which is still inside the go routine. Maybe use a errCh (e.g. https://github.com/kubernetes/kubernetes/blob/64a6ee4321d37740b84a5c5c7aaa67e0bf6b6508/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection_test.go#L123)? The select could be combined with the below `<-done`

Reply:
> 该`Fatalf`呼叫退出早，但`Errorf`没有，所以如果将打印错误，然后有将在线路61零指针异常，这仍然是旅途例程中。也许使用errCh（例如
> 
> https://github.com/kubernetes/kubernetes/blob/64a6ee4321d37740b84a5c5c7aaa67e0bf6b6508/staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/connection_test.go#L123
> 
> ）？选择内容可以与以下内容结合使用`<-done`

This sounds more reasonable and I want to continue to fix it